ssl/ssl_lib.cc

```diff
@@ -2476,16 +2485,21 @@ EVP_PKEY *SSL_get_privatekey(const SSL *ssl) {
     assert(ssl->config);
     return NULL;
   }
-  if (ssl->config->cert != NULL) {
-    return ssl->config->cert->privatekey.get();
+  if (ssl->config->cert != NULL &&
+      ssl_cert_check_cert_private_keys_usage(ssl->config->cert.get())) {
```

Comment:
If the null check is added to `ssl_cert_check_cert_private_keys_usage`, then it's not needed here.